app/models/adviser.rb

```diff
@@ -29,6 +29,27 @@ class Adviser < ActiveRecord::Base
 
   validate :match_reference_number
 
+  after_save :geocode_if_needed
+
+  def full_street_address
+    [postcode, 'United Kingdom'].delete_if(&:blank?).join(', ')
```

Comment:
We can just interpolate postcode with United Kingdom here as we know by this point we'd be valid and postcode would never be blank.